### DISCUSSION TITLE
Setting up the `BugBug` repository

### DISCUSSION THREAD

**Discussion by jpangas** [mozilla/bugbug#3338] [Q&A] [answered]
How can I setup the `bugbug` repository  on my computer?

**Comment by srishtig2412** [comment]
I have made the changes. Kindly, let me know if any further changes are required

**Reply by jpangas** [reply]
Hi @srishtig2412 , thanks for the changes. @suhaibmujahid recommended we place the full guide/steps in this discussions section. What you can do is, include a sentence or two in `CONTRIBUTION.md` that link to this discussion incase anyone wants further guidelines on how to contribute.. I think it is better to have the steps here because of added functionalities and if anyone gets a problem, they can also comment below.

**Reply by srishtig2412** [reply]
Sure, I have added the discussions link at the end. I hope is this fine?

**Comment by srishtig2412** [comment]
Hello, Actually I have recently started open source contributions and currently learning for the same. Could you help me with suggesting some more beginner-friendly projects to contribute for in this project ??

**Reply by jpangas** [reply]
For a beginner in bugbug, [Issues marked as good-first-bug](https://github.com/mozilla/bugbug/labels/good-first-bug) are self-contained enough that a contributor should be able to work on them.
All Issues are considered not assigned, until there is a PR linked to them. Feel free to work on any unassigned issue, you don't need to ask first.

**Reply by jpangas** [reply]
Feel free to pick up any `good-first-bug` issue and work on it. They are all beginner friendly.

**Comment by jpangas** [comment, marked as answer]
## Setting Up the repository

1. Create a fork of the project by clicking the `Fork` button near the top of the page.This will create a personal copy of the repository on your Github account.

2. Clone the forked repository from your GitHub account onto your machine:

```bash
git clone git@github.com:YourGitUsername/bugbug.git
```

- 2.5  Follow the [**Setups and Prerequisites**](https://github.com/mozilla/bugbug#setup-and-prerequisites) steps in `README.md`

3. Add the upstream remote repository. This will save a reference which can be used to synchronize changes later.
```bash
git remote add upstream git@github.com:mozilla/bugbug.git
```

4. Run the command `git remote -v` to confirm the aliases for both the origin and remote repository are set and should display:
<pre>
origin  https://github.com/YourGitUsername/bugbug.git (fetch)
origin  https://github.com/YourGitUsername/bugbug.git (push)
upstream        https://github.com/mozilla/bugbug.git (fetch)
upstream        https://github.com/mozilla/bugbug.git (push)
</pre>
5.  At this stage, the project should be working on your machine and your git repository has been fully configured to make your first contribution. Please also see #3342 to learn how you can make your first contribution.

**Reply by fadebowaley** [reply]
Thank you for this!
If  you choose option of **setting your virtual enviroment**, before installing  make sure you add it to your  .gitignore file for best pracices
